### ISSUE TITLE
Variables in handlebars partials are cached

### ISSUE THREAD

**Issue by Bessonov** [webdiscus/html-bundler-webpack-plugin#128]
## Current behaviour

I see a caching issue: https://github.com/Bessonov/handlebars-inheritance/commit/1f0c87fecdc290f4c914956faad6619498e1a5bc

Basically, templates:
```handlebars
// layout / parent
{{#if a}}a{{/if}}
{{#if b}}b{{/if}}

// index / child
{{#> layout}}
{{/layout}}
```

```js
index({ a: true }) // outputs as expected "a"
index({ b: true }) // outputs "a b"
```

## Expected behaviour

```js
index({ a: true }) // outputs "a"
index({ b: true }) // outputs "b"
```

### Reproduction

- https://github.com/Bessonov/handlebars-inheritance/

## Environment

- OS: Linux
- version of Node.js: v22.11.0
- version of Webpack: 5.97.0
- version of the Plugin: 4.7.8

## Appreciation for the useful project

- [⭐] After the problem is resolved, do not forget to give a star ⭐

**Comment by Bessonov** [comment]
Update: It is even simpler, without inheritance:

```handlebars
{{#if a}}a{{/if}}
{{#if b}}b{{/if}}
```
```js
index({ a: true }) // outputs as expected "a"
index({ b: true }) // outputs "a b"
```

https://github.com/Bessonov/handlebars-inheritance/blob/d37bd14cbe26b981d4871e88feec49677cc2a94e/src/index.hbs

**Comment by webdiscus** [comment]
@Bessonov 

Thanks for the report of the issue. I will fix it.

**Comment by webdiscus** [comment]
@Bessonov 

the issue is fixed in `v4.8.1`

Thank you!

**Comment by Bessonov** [comment]
Thanks, @webdiscus, the change seems to be working!

Viele Grüße aus Recklinghausen!
